enumerables.rb

```diff
@@ -0,0 +1,261 @@
+# Our enumerable methods
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/MethodLength
+# rubocop:disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    arr = self
+    if block_given?
+      arr = arr.is_a?(Array) ? arr : arr.to_a
+      arr.length.times { |i| yield(arr[i]) }
+    end
+    arr.to_enum
+  end
+
+  def my_each_with_index
+    arr = self
+    if block_given?
+      arr = arr.is_a?(Array) ? arr : arr.to_a
+      arr.length.times { |i| yield(arr[i], i) }
+    end
+    arr.to_enum
+  end
+
+  def my_select
+    arr = self
+    if block_given?
+      new_arr = []
+      arr = arr.is_a?(Array) ? arr : arr.to_a
+      arr.my_each { |condition| new_arr.push(condition) if yield(condition) }
+      new_arr
+    end
+    arr.to_enum
```

Comment:
### Fix the error and test to pass the case

Reply:
Implemented this